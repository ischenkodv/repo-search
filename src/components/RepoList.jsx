import React from 'react';
 
class RepoList extends React.Component {
  render() {
    if (!Array.isArray(this.props.repositories)) {
      return null;
    }

    return (
      <ul className="list-group">{
        this.props.repositories.map(repository => {
          return <li key={ 'repo_' + repository.name } className="list-group-item">
            <a href={repository.html_url} target="_blank">{ repository.name }</a>
          </li>
        })
      }</ul>
    );
  }
}

export default RepoList;
