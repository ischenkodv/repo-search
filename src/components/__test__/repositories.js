module.exports = [
  { name: 'test 1', html_url: 'https://example.com/test1' },
  { name: 'test 2', html_url: 'https://example.com/test2' },
  { name: 'test 3', html_url: 'https://example.com/test3' },
];
