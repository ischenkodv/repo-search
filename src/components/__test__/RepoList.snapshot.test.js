import RepoList from '../RepoList';
import renderer from 'react-test-renderer';
import repositories from './repositories';
import React from 'react';

test('RepoList renders the list of repositories', () => {
  const rendered = renderer.create(
    <RepoList repositories={repositories} />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
