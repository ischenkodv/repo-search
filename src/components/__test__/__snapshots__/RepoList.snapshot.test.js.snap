exports[`test RepoList renders the list of repositories 1`] = `
<ul
  className="list-group">
  <li
    className="list-group-item">
    <a
      href="https://example.com/test1"
      target="_blank">
      test 1
    </a>
  </li>
  <li
    className="list-group-item">
    <a
      href="https://example.com/test2"
      target="_blank">
      test 2
    </a>
  </li>
  <li
    className="list-group-item">
    <a
      href="https://example.com/test3"
      target="_blank">
      test 3
    </a>
  </li>
</ul>
`;
